### DISCUSSION TITLE
Shifting "gjs-toolbar-items" div to the bottom-right of the component instead of top-right

### DISCUSSION THREAD

**Discussion by zixiang2018** [GrapesJS/grapesjs#4948] [Q&A] [answered]
Hi community,

Hope you are having an great day!

May I check, how can I shift the blue toolbar (the toolbar with className `gjs-toolbar-items`) of a selected component to the bottom-right instead of its default position (top-right of the component)?
<img width="358" alt="Screenshot 2023-02-27 at 6 17 18 PM" src="https://user-images.githubusercontent.com/54768054/221537393-7c17a7c4-1244-4441-860a-22f3839adfa1.png">

Did my research, I have seen discussions on how to set the contents of the `gjs-toolbar-items` by modifying the component `model.defaults.toolbar` but I couldn't find any discussion about modifying the position of the blue toolbar relative to the selected component.

Thank you for reading and any help is appreciated! 🙇

**Comment by artf** [comment, marked as answer]
One quick fix would be to force its position via CSS in this way
```css
.gjs-toolbar {
    bottom: -26px; /* <- toolbar height */
    top: auto !important;
}
```
but that would prevent its dynamic repositioning logic (eg. when the toolbar goes out of the screen).

If you want to go a bit deeper you could read the [command responsible for that logic](https://github.com/GrapesJS/grapesjs/blob/45f23acf975d8ab09b60c095a3b405002b8c1658/src/commands/view/SelectComponent.ts#L511-L517) and [update it](https://grapesjs.com/docs/modules/Commands.html#extending) with your own.
